### PR TITLE
[Core] Add Checklist and AbilityRequirement types

### PR DESCRIPTION
**Pull request by Pewtro**

### CODE REVIEW

File: src/parser/monk/brewmaster/modules/core/Checklist/Component.tsx [hunk 1]
```diff
@@ -7,17 +7,18 @@ import Rule, { PERFORMANCE_METHOD } from 'parser/shared/modules/features/Checkli
...
-const Component = ({ combatant, castEfficiency, thresholds }: any) => {
-  const AbilityRequirement = (props: any) => (
+const Component = ({ combatant, castEfficiency, thresholds }: ChecklistProps) => {
+  const AbilityRequirement = (props: AbilityRequirementProps) => (
     <GenericCastEfficiencyRequirement
       castEfficiency={castEfficiency.getCastEfficiencyForSpellId(props.spell)}
       {...props}
     />
   );
   AbilityRequirement.propTypes = {
```
Comment (Dambroda) [inline src/parser/monk/brewmaster/modules/core/Checklist/Component.tsx]: should this be gone?

File: src/parser/shared/modules/features/Checklist/ChecklistTypes.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export interface ChecklistProps {
+  combatant: Combatant,
+  castEfficiency: CastEfficiency,
+  thresholds: { [name: string]: RequirementThresholds }
+}
+
+export interface AbilityRequirementProps {
+  spell: number,
+  name?: string | JSX.Element,
```
Comment (Dambroda) [inline src/parser/shared/modules/features/Checklist/ChecklistTypes.ts:13]: is there an example of where this is used? or can we just remove it?

**Review by Dambroda — APPROVED** [review]
Looks good to me. Two comments/questions below, and then also one overarching comment: I noticed that you removed `AbilityRequirement.proptypes = {}`, but the actual `checklist.proptypes` are still in are the bottoms of the files. Was that intentional?

File: src/parser/shared/modules/features/Checklist/ChecklistTypes.ts (hunk 2 above)
Comment (Pewtro) [reply src/parser/shared/modules/features/Checklist/ChecklistTypes.ts:13]: Once... 😓  in Brewmaster: 
``<AbilityRequirement spell={SPELLS.CELESTIAL_BREW.id} name={<><SpellLink id={SPELLS.CELESTIAL_BREW.id} /> cast efficiency</>} />
``

File: src/parser/monk/brewmaster/modules/core/Checklist/Component.tsx (hunk 1 above)
Comment (Pewtro) [reply src/parser/monk/brewmaster/modules/core/Checklist/Component.tsx]: Yes!

**Comment by Pewtro** [comment]
No that was not intentional - fixed in latest PR.

**Comment by Dambroda** [comment]
> No that was not intentional - fixed in latest PR.

Everything I noticed is now resolved :D

**Pull request merged**